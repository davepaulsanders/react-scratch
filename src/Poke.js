import React, { useState, useEffect } from "react";
import { useQuery } from "react-query";
import { Pages } from "./Pages";
export const Poke = () => {
  const [pageURL, setPageURL] = useState("https://pokeapi.co/api/v2/pokemon");
  const [pokemon, setPokemon] = useState([]);
  useEffect(() => {
    try {
      fetchPokemon();
    } catch (err) {
      console.log(err);
    }
  }, []);
  const fetchPokemon = async () => {
    try {
      const poke = await fetch("https://pokeapi.co/api/v2/pokemon");
      const pokeJSON = await poke.json();
      console.log(pokeJSON)
      setPokemon(pokeJSON.results);
      // return pokeJSON;
    } catch (err) {
      console.log(err);
    }
  };
  // const { isLoading, error, data } = useQuery(
  //   ["getPokemon", pageURL],
  //   () => fetchPokemon(pageURL),
  //   { keepPreviousData: true }
  // );

  // if (isLoading) {
  //   return <div>Loading...</div>;
  // }
  // if (error) {
  //   return <div>Something went wrong</div>;
  // }
  if (pokemon) {
    console.log(pokemon);
    return (
      <div className="d-flex align-items-center flex-column">
        <div className="d-flex justify-content-center w-50 flex-wrap p-5">
          {pokemon.map((poke) => (
            <p id= "poke" className="border border-dark text-center p-4" key={poke.name}>
              {poke.name}
            </p>
          ))}
        </div>
        <Pages pokemonData={pokemon} setPageURL={setPageURL} />
      </div>
    );
  }
};
