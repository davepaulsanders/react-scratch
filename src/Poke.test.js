import React from "react";
import renderer from "react-test-renderer";
import {render} from "@testing-library/react"
import { Poke } from "./Poke";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([{ name: "Bulbasaur" }, { name: "Venasaur" }]),
  })
);

it("matches the snapshot", () => {
  const element = renderer.create(<Poke />);
  let tree = element.toJSON();
  expect(tree).toMatchSnapshot();
});

it("fetches data", () => {
  const element = renderer.create(<Poke />);
  expect(fetch).toHaveBeenCalledTimes(1);
});

it("lists pokemon", () => {
  const {container} = render(<Poke />);
  
  expect(container.getById("poke")).toBe("Bulbasaur");
});
