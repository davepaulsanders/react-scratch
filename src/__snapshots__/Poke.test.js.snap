// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="d-flex align-items-center flex-column"
>
  <div
    className="d-flex justify-content-center w-50 flex-wrap p-5"
  />
  <div>
    <button
      className="btn btn-primary m-2"
      disabled={false}
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="btn btn-primary m-2"
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
